src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -165,6 +191,16 @@ def getPileup(self, spec, projection=None):
         """
         results = []
 
+        # remove API key from the spec as it defines pileup API
+        if isinstance(spec, dict) and 'API' in spec:
+            spec.pop('API')
+
+        self.logger.info("getPileup spec=%s, projection=%s", spec, projection)
+        err = self.validate(spec, projection)
+        if len(err) != 0:
+            self.logger.error("getPileup validation error %s", err)
```

Comment:
I'd suggest to rephrase it to something like "Get pileup query didn't pass the validation. Error: %s"

Reply:
done